### ISSUE TITLE
Remove "arrayProp" & "mapProp" decorator

### ISSUE THREAD

**Issue by hasezoey** [typegoose/typegoose#258]
(in addition to #257)
Remove `@arrayProp` & `@mapProp` decorators in favor of auto-detection

solution:
```ts
// normal prop
@prop({ type: () => String })
public someprop: string; // detected as non-map & non-array
// design:type -> String
// array prop
@prop({ type: () => [String] })
public somearrayprop: string[]; // detected as array
// design:type -> Array
// map prop
@prop({ type: () => String })
public somemapprop: Map<string, string>; // detected as map
// design:type -> Map
```

proposed by @andreialecu on discord

**Comment by hasezoey** [comment]
since e7a9f5e146f14b2ca987d524cd3bcb307ef6e1f2 they are just aliases for `@prop` and have an TSDoc `@deprecated`

**Comment by hasezoey** [comment]
closing, because released on beta since `8.0.0-beta.1`
